pgraph/actions.go

```diff
@@ -306,6 +306,149 @@ func (obj *SentinelErr) Error() string {
 	return obj.err.Error()
 }
 
+// innerWorker is the CheckApply runner that reads from processChan.
+func (g *Graph) innerWorker(v *Vertex) {
+	obj := v.Res
+	running := false
+	done := make(chan struct{})
+	playback := false                      // do we need to run another one?
+	_, wcuid, pcuid := obj.ConvergerUIDs() // get extra cuids (worker, process)
+
+	waiting := false
+	var timer = time.NewTimer(time.Duration(math.MaxInt64)) // longest duration
+	if !timer.Stop() {
+		<-timer.C // unnecessary, shouldn't happen
```

Comment:
Great question! Actually the timer docs seem to say that this is needed, but in practice it never seems to happen. So your guess is as good as mine if I'm using timer correctly, but it does seem to work as expected.